src/arrayMethodReduce.js

```diff
@@ -5,7 +5,19 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let start = this[0];
+    let sameShitAsIndex = 1;
+
+    if (arguments.length === 2) {
+      start = initialValue;
+      sameShitAsIndex = 0;
+    }
+
+    for (let index = sameShitAsIndex; index < this.length; index++) {
```

Comment:
for better understanding of the code, it is a "best practice" to declare the index in the body of the loop. We improve the code style so that our colleagues can understand it and quickly understand other people's (our) code